de.tudarmstadt.ukp.wikipedia.api/src/main/java/de/tudarmstadt/ukp/wikipedia/api/CategoryGraph.java

```diff
@@ -52,14 +51,17 @@
 import de.tudarmstadt.ukp.wikipedia.util.ApiUtilities;
 import de.tudarmstadt.ukp.wikipedia.util.CommonUtilities;
 import de.tudarmstadt.ukp.wikipedia.util.OS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * The category graph is constructed from the links connecting Wikipedia categories.
  * It provides various accessors and graph algorithms.
  *
  */
 public class CategoryGraph implements WikiConstants, Serializable {
-	private final Log logger = LogFactory.getLog(getClass());
+    
+    private static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
```

Comment:
That looks interesting :)

Reply:
It's taken from the official FAQ (https://www.slf4j.org/faq.html#declaration_pattern) to avoid CnP errors with hard-coded Class names. The recommended idiom "can be cut and pasted across classes" without introducing errors en passant.